### PR TITLE
Rewrite of the getting started introduction

### LINKED ISSUE #4035: Rewrite of the Getting Started section: plan
The task: Rewrite the getting started guide to explain key concepts of Godot and help users get started in a couple of hours. The content should frequently link to other pages in the user manual for more information.

Notes:

_1. By rewrite, I do not mean to throw away everything and restart from zero. But there is a lot of writing and editing work involved with this task.
2. Multiple sub-sections in the current Getting Started section are going to move to other sections. In general, we want Getting started to be one short series, and then link people to other parts of the docs they want to explore. See #3969 for more information._

## Introduction and goals

Godot has different kinds of users with varied skills and needs. The current step-by-step guide contains mixed content, sometimes with language that's too technical for many users. As many persons wrote it over several years, it also lacks coherence.

This task aims to make the series accessible enough so people who are new to game creation, but not entirely new to programming, can follow it. Having some programming foundations is a pre-requisite to learn an extensive API like Godot's.

For people who have no programming experience, we will link to comprehensive free resources to get them started:

1.  Harvard's intro to computer science, CS50, a [free open courseware](https://cs50.harvard.edu/x/2020/), and relies on open-source technologies.
2.  The free ebooks and Udemy courses by Al Sweigart are great for self-paced learning: [Automating the boring stuff with Python](https://automatetheboringstuff.com/), and Invent with Scratch for young adults.

The getting started guide should not cover engine features in-depth. Instead, it will link to pages that cover given topics in greater depth.

The getting started section should help users:

1.  Understand the fundamental concepts of Godot, like nodes, scripting, the scene tree, and more.
2.  Get a sense of available features, how they work, and see if Godot is a good match for their projects.
3.  Know their way around the editor.
4.  Get started creating 2D and 3D games with Godot.
5.  Find more information about specific aspects of the engine.

## The plan

The plan below is a list of topics the Getting Started section should cover. The order and topics are subject to change, of course. This redesign of the getting started series is meant to make it more coherent and to use consistent writing and language.

The Getting Started section page should explain how the guide works and the pre-requisites for each series. In particular, it should link beginners to free resources to learn to code.

I'm also going to work on a landing page for the docs linking to other sections as a separate task.

1.  Introduction
    1.  Introduction to Godot and general overview its features: a guide to help answer questions such as "Is Godot for me? What can it do for me?"
    2.  First look at the editor and its interface.
    3.  Overview of Godot's main concepts: brushing over the scene nodes, scenes, scripts, scene tree, signals, in a short page. Unlike the current pages about scripting, instancing, etc., the goal is to give you a sense of how Godot works in a few minutes.
    4.  Learning new features: a guide to using the built-in code reference, online documentation, and best practices to get help on available communities.
2.  Nodes and Scenes: a longer introduction to nodes, their features, and composing them into reusable scenes
3.  Creating your first scene
4.  Introduction to scripting: overview of available programming languages
5.  GDScript hands-on: a non-exhaustive, brief, but hands-on introduction to GDScript, adding behavior to a scene.
6.  Reacting to changes with signals: a short intro to signals, and using them in the previously created scene.
7.  Creating instances of a scene: intro to instancing, changing properties in the original scene, modifying the instance
8.  Creating your first 2D Game (already available, Dodge the Creeps)
9.  Creating your first 3D Game

## Moving or removing existing content

I would move some existing content in the Step-by-step series to other places, and link to it from the getting started series:

1.  Signals documentation could be in Scripting.
2.  Exporting is covered in Project Workflow, I'd keep it all there and link to the corresponding section.
3.  File System to either Scripting or Input and Output.
4.  Resources: we could brush over the basics of resources in Getting Started, explaining that all images etc. you put into nodes are resources. The page contains more technical and reference-like information that should move to… Scripting?
5.  The intro to animation should be in Animation. It doesn't need to be a tutorial about animating a splash screen.
6.  SceneTree is a technical article that explains Godot's internals and initialization.
7.  Singletons -\> Scripting

To remove:

1.  GUI tutorials
2.  Splash screen tutorial

## To review and integrate

<https://github.com/godotengine/godot-docs/pull/2958/files>

### PR DESCRIPTION
**Pull request by NathanLovato**
This PR covers the introduction to the new getting started section. It's an introduction to Godot that:

1. Answers common questions and clarifies pre-requisites for the users.
2. Gives a quick overview of key concepts that are nodes, scenes, the scene tree, and signals. This page doesn't try to be exhaustive but only introduce what's essential in the next parts of the series.
3. Gives an overview of the editor's interface. This isn't a step-by-step guide to using the editor, just a quick intro. Later in the getting started series, we'll cover usage as we introduce new parts of the editor. For detailed usage and extras like customizing the editor, we should instead cover that in detail in the editor/ section.
4. Explains how to use available resources to learn Godot. Thanks to @willnationsdev for writing that page. There, we explain how to use available docs, provide two free programming courses to beginners (Harvard CS50 and automating the boring stuff with Python), and explain how to use community channels to ask questions.

See #4035 for more information about the complete rewrite plan.

Individual commits close corresponding issues.

### CODE REVIEW

**Comment by NathanLovato** [comment]
I included some comments at the top of the rst files to explain the pages' teaching goals to future contributors. Hopefully, it'll help to avoid putting too much information in individual pages, especially introductory ones. I also included a comment about missing content in the page about customizing the editor; I'm not writing it myself because it's beyond the scope of the current job.

**Comment by NathanLovato** [comment]
Thanks @willnationsdev. Would you like write access to my fork, so you can directly fix typos and commit without having to write individual comments?

**Comment by willnationsdev** [comment]
@NathanLovato that is fine with me, if you're willing.

**Comment by NathanLovato** [comment]
Absolutely, I'm always happy to work more directly with people! Just sent you access.

File: getting_started/introduction/first_look_at_the_editor.rst [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+console, the animation editor, the audio mixer, and more. They can take precious
+space, that's why they're folded by default.
+
+.. image:: img/editor_intro_bottom_panels.png
+
+When you click on one, it expands vertically. Below, you can see the animation editor opened.
+
+.. image:: img/editor_intro_bottom_panel_animation.png
+
+The four workspaces
```
Comment (aaronfranke) [inline getting_started/introduction/first_look_at_the_editor.rst:94]: These are called "main screens" elsewhere in the documentation. I'm not against "workspaces", but I think we should stick to one name and use it consistently in the documentation.

File: getting_started/introduction/first_look_at_the_editor.rst [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+--------------------------
+
+Godot comes with a built-in class reference.
+
+You can search for information about a class, method, property, constant, or
+signal by:
+
+1. Pressing :kbd:`Shift + F1` anywhere in the editor.
+2. Clicking the "Search Help" button in the top-right of the Script workspace.
+3. Clicking on the Help menu and Search.
```
Comment (aaronfranke) [inline getting_started/introduction/first_look_at_the_editor.rst:143]: You can also Ctrl+Click on an API of some kind.

File: getting_started/introduction/introduction_to_godot.rst [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+How does it work and look?
+--------------------------
+
+Godot comes with a fully-fledged game editor with integrated tools to answer the
+most common needs. It includes a code editor, an animation editor, a tilemap
+editor, a shader editor, a debugger, a profiler, and more.
+
+.. image:: img/introduction_editor.png
+
+The team strives to offer a complete game creation package with a consistent
```
Comment (aaronfranke) [inline getting_started/introduction/introduction_to_godot.rst:59]: This is misleading (or wrong). Godot is not a complete game creation tool unto itself, but rather another free component of a fully functioning open source game development ecosystem made useful by an image editor like GIMP or Krita, 3D modeling software such as Blender, and more.

File: getting_started/introduction/key_concepts_overview.rst [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+.. image:: img/key_concepts_character_nodes.png
+
+It is made of a ``KinematicBody2D`` node named "Character", a ``Sprite2D``, a
+``Camera2D``, and a ``CollisionShape2D``.
+
+.. note:: The node names end with "2D" because this is a 2D scene. Their 3D
+          counterpart have names that end with "3D".
+
+Notice how nodes and scenes look the same in the editor. When you save a tree of
+nodes as a scene, it then acts as a single node, hiding its internal structure.
```
Comment (aaronfranke) [inline getting_started/introduction/key_concepts_overview.rst:50]: This is misleading. The structure isn't really hidden, you can still get to the children in the editor by enabling "Editable Children", and you can get to the children in the code by using `get_child`/`get_node`/etc. Instanced scenes are also still very visible due to the clapperboard icon.

Comment (willnationsdev) [reply getting_started/introduction/key_concepts_overview.rst:50]: But the intended usage of the design, from an Object Orientation standpoint, is as stated, so I think as an overview of how it should work / be perceived by users, this is sufficient. You can always dig into the details of how to subvert the default state in-editor/at runtime in later docs.

Comment (aaronfranke) [reply getting_started/introduction/key_concepts_overview.rst:50]: I agree @willnationsdev, but we should still try to phrase things such that it's neither misleading nor confusing.

How about "it then shows up as a single node, with its internal structure hidden in the editor"?

**Comment by NathanLovato** [comment]
Thanks for the comments, I'll finish writing the 3D tut and get to all your notes later!

File: getting_started/introduction/introduction_to_godot.rst (hunk 3 above)
Comment (NathanLovato) [reply getting_started/introduction/introduction_to_godot.rst:59]: Meant it in the sense of full-featured game engine with editors for most tasks (script editor, tileset, anim, anim tree, profiler, etc.). Editing this now.

**Comment by NathanLovato** [comment]
Thanks lads, just addressed your comments.

File: getting_started/introduction/first_look_at_the_editor.rst [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+When you click on one, it expands vertically. Below, you can see the animation editor opened.
+
+.. image:: img/editor_intro_bottom_panel_animation.png
+
+The four main screens
+-------------------
+
+There are four main screen buttons centered at the editor's top: 2D, 3D, Script,
+and AssetLib.
```
Comment (aaronfranke) [inline getting_started/introduction/first_look_at_the_editor.rst:98]:
```suggestion
---------------------

There are four main screen buttons centered at the top of the editor:
2D, 3D, Script, and AssetLib.
```

File: getting_started/introduction/first_look_at_the_editor.rst [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+Godot comes with a built-in class reference.
+
+You can search for information about a class, method, property, constant, or
+signal by:
+
+1. Pressing :kbd:`Shift + F1` anywhere in the editor.
+2. Clicking the "Search Help" button in the top-right of the Script main screen.
+3. Clicking on the Help menu and Search.
+4. Clicking while pressing the :kbd:`Ctrl` key on a built-in variable or
+   function name in the script editor.
```
Comment (aaronfranke) [inline getting_started/introduction/first_look_at_the_editor.rst:145]:
```suggestion
4. Clicking while pressing the :kbd:`Ctrl` key on a class name,
   function name, or built-in variable in the script editor.
```

**Comment by NathanLovato** [comment]
Thanks!

**Comment by akien-mga** [comment]
I didn't have time to do a review yet, but others did (thanks!). I don't want to block Nathan too long so given that I trust the general direction he's taking, let's merge and see what changes need to be made based on users' feedback.

I'll still try to review post-merge to see if there's changes I'd like to make (and if so I might do a PR directly).

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**